app/src/tx_display.c

```diff
@@ -121,6 +121,13 @@ __Z_INLINE parser_error_t calculate_is_default_chainid()
     return parser_ok;
 }
 
+bool has_testnet_chain_id(const char *chain_id)
+{
+    size_t len_prefix = strlen(TESTNET_CHAINID_PREFIX),
+           len_chain_id = strlen(chain_id);
+    return len_chain_id < len_prefix ? false : strcmp(TESTNET_CHAINID_PREFIX, chain_id) == 0;
```

Comment:
`strcmp` will check for an exact match, returning non-zero if the chain_id is `kava-testnet-8000` since `kava-testnet` has a null termination while chain_id has more characters.

We need to use `strncmp` here and only check the first characters up to the prefix length.  It's probably worth adding some test cases for this function too.